p2p/discover/table.go

```diff
@@ -541,6 +547,9 @@ func (tab *Table) addVerifiedNode(n *node) {
 	if tab.nodeAddedHook != nil {
 		tab.nodeAddedHook(n)
 	}
+	if metrics.Enabled {
+		bucketsCounter[b.index].Inc(1)
+	}
```

Comment:
Maybe add functions `nodeAddedMetrics`, `nodeRemovedMetrics` to do this.

Reply:
I reused `nodeAddedHook` and added a matching `nodeRemovedHook`. This felt a little better since we already had half of the functions, even if it was originally for testing. I made a new function `newMeteredTable` so it is clearer to caller that the shouldn't override `node{Added|Removed}Hook`.